src/components/buttons/Button/Button.tsx

```diff
@@ -0,0 +1,41 @@
+import { ReactNode } from 'react';
+import classNames from 'classnames/bind';
+import {
+  ButtonColorType,
+  ButtonRadiusType,
+  ButtonWidthType,
+  BUTTON_COLOR,
+  BUTTON_WIDTH,
+  BUTTON_RADIUS,
+} from '@/constants/buttonTypes';
+import styles from './Button.module.scss';
+
+const cn = classNames.bind(styles);
+
+interface ButtonProps {
+  backgroundColor?: ButtonColorType;
+  hoverColor?: ButtonColorType;
+  radius?: ButtonRadiusType;
+  width?: ButtonWidthType;
+  onClick: () => void;
+  children: ReactNode;
+}
+
+export default function Button({
+  backgroundColor = BUTTON_COLOR.BACKGROUND_MAIN_01,
+  hoverColor,
+  radius = BUTTON_RADIUS.RADIUS_0,
+  width = BUTTON_WIDTH.PARENT_FULL,
+  onClick,
+  children,
+}: ButtonProps) {
+  return (
+    <button
+      className={cn('common-style', `${backgroundColor}`, `${radius}`, `${width}`, `hover-${hoverColor}`)}
```

Comment:
```suggestion
      className={cn('common-style', backgroundColor, radius, width, `hover-${hoverColor}`)}
```
요렇게 적는 방법도 있을거 같아용~!